lib-ee/emqx_ee_schema_registry/src/emqx_ee_schema_registry.erl

```diff
@@ -0,0 +1,242 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2023 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%--------------------------------------------------------------------
+-module(emqx_ee_schema_registry).
+
+-behaviour(gen_server).
+-behaviour(emqx_config_handler).
+
+-include("emqx_ee_schema_registry.hrl").
+-include_lib("emqx/include/logger.hrl").
+-include_lib("snabbkaffe/include/snabbkaffe.hrl").
+
+%% API
+-export([
+    start_link/0,
+
+    get_serde/1,
+
+    add_schema/2,
+    delete_schema/1,
+    list_schemas/0
+]).
+
+%% `gen_server' API
+-export([
+    init/1,
+    handle_call/3,
+    handle_cast/2,
+    terminate/2
+]).
+
+%% `emqx_config_handler' API
+-export([post_config_update/5]).
+
+-type schema() :: #{
+    type := serde_type(),
+    source := binary(),
+    description => binary()
+}.
+
+%%-------------------------------------------------------------------------------------------------
+%% API
+%%-------------------------------------------------------------------------------------------------
+
+start_link() ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+-spec get_serde(schema_name()) -> {ok, serde_map()} | {error, not_found}.
+get_serde(SchemaName) ->
+    case ets:lookup(?SERDE_TAB, to_bin(SchemaName)) of
+        [] ->
+            {error, not_found};
+        [Serde] ->
+            {ok, serde_to_map(Serde)}
+    end.
+
+-spec add_schema(schema_name(), schema()) -> ok | {error, term()}.
+add_schema(Name, Schema) ->
+    RawSchema = emqx_map_lib:binary_key_map(Schema),
+    Res = emqx_conf:update(
+        [?CONF_KEY_ROOT, schemas, Name],
+        RawSchema,
+        #{override_to => cluster}
+    ),
+    case Res of
+        {ok, _} ->
+            ok;
+        Error ->
+            Error
+    end.
+
+-spec delete_schema(schema_name()) -> ok | {error, term()}.
+delete_schema(Name) ->
+    Res = emqx_conf:remove(
+        [?CONF_KEY_ROOT, schemas, Name],
+        #{override_to => cluster}
+    ),
+    case Res of
+        {ok, _} ->
+            ok;
+        Error ->
+            Error
+    end.
+
+-spec list_schemas() -> #{schema_name() => schema()}.
+list_schemas() ->
```

Comment:
Nit: This is actually a map 😅

Reply:
`list` here is not representing the return type, but rather the action ("list the existing schemas") :smile_cat: